src/corona_hakab_model/generation/geographic_circle.py

```diff
@@ -34,22 +36,34 @@ def generate_agents_ages_and_connections_types(self):
         Iterates over each agent. Generates it's age by a given age distribution (self.data_holder.age_distribution).
         For each agent, iterates over each connection type, and rolls whether it has this type of connection or not.
         Gets this probabilities from self.data_holder.connection_types_prob_by_age.
-        It allows choosing whether some one goes to work, to school, or to none of those.
+        It allows choosing whether some one goes to work, to school, to kindergarten or to none of those.
         :return:
         """
         ages = iter(self.data_holder.age_distribution.rvs(size=len(self.agents)))
+        workplace_distribution = rv_discrete(
+            values=([ConnectionTypes.School, ConnectionTypes.Kindergarten, ConnectionTypes.Work],
+                    [self.data_holder.teachers_workforce_ratio, self.data_holder.kindergarten_workforce_ratio,
+                    1 - self.data_holder.teachers_workforce_ratio - self.data_holder.kindergarten_workforce_ratio])
+        )
         for agent, age in zip(self.agents, ages):
             agent.age = age
             agent_connection_types = []
-            for connection_type in ConnectionTypes:
+            for connection_type in Non_Exclusive_Types:
                 if np.random.random() < self.data_holder.connection_types_prob_by_age[age][connection_type]:
                     agent_connection_types.append(connection_type)
             
-            if agent.is_adult() and ConnectionTypes.Work in agent_connection_types:
-                # if the agent works, there's a chance it is a teachers job 
-                if np.random.random() < self.data_holder.teachers_workforce_ratio:
-                    agent_connection_types.remove(ConnectionTypes.Work)
-                    agent_connection_types.append(ConnectionTypes.School)
+            if agent.is_adult() and np.random.random() < self.data_holder.connection_types_prob_by_age[age][
+                ConnectionTypes.Work]:
+
+                # if the agent works, decide workplace
+                agent_connection_types.append(workplace_distribution.rvs())
+
+            education_prob = [self.data_holder.connection_types_prob_by_age[age][connection_type] for
+                              connection_type in Education_Types]
+            education_type = rv_discrete(values=([Education_Types + [-1], education_prob + [1 - sum(education_prob)]])).rvs()
```

Comment:
instead of the "+ [-1]", why not compare random() with sum(education_prob) in the IF below? much more readable

Reply:
That was my original implementation. This will require normalizing the probabilities, but I guess it is more readable